### PR TITLE
Advance velox version and use consolidated memory pool

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Also fix the build breakage caused by
https://github.com/facebookincubator/velox/pull/3137

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/TaskResource.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class TaskResource {
       const std::vector<std::string>& pathMatch);
 
   TaskManager& taskManager_;
-  std::unique_ptr<velox::memory::ScopedMemoryPool> pool_;
+  std::shared_ptr<velox::memory::MemoryPool> pool_;
```
Comment (spershin) [inline presto-native-execution/presto_cpp/main/TaskResource.h:68]: Why do we have shared instead of unique pointers now?

**Comment by spershin** [comment]
@xiaoxmeng Format check failed
```
-  auto pool = memory::getProcessDefaultMemoryManager().getRoot().addChild(
-      "query_root");
+  auto pool =
+      memory::getProcessDefaultMemoryManager().getRoot().addChild("query_root");
   pool->setMemoryUsageTracker(velox::memory::MemoryUsageTracker::create(
       maxUserMemoryPerNode, maxSystemMemoryPerNode, maxTotalMemoryPerNode));
```

File: presto-native-execution/presto_cpp/main/TaskResource.h (hunk 1 above)
Comment (xiaoxmeng) [reply presto-native-execution/presto_cpp/main/TaskResource.h:68]: The child memory pool holds a shared reference to its parent. So we need to use shared ptr to maintain the memory pool hierarchy.

**Comment by xiaoxmeng** [comment]
> @xiaoxmeng Format check failed
> 
> ```
> -  auto pool = memory::getProcessDefaultMemoryManager().getRoot().addChild(
> -      "query_root");
> +  auto pool =
> +      memory::getProcessDefaultMemoryManager().getRoot().addChild("query_root");
>    pool->setMemoryUsageTracker(velox::memory::MemoryUsageTracker::create(
>        maxUserMemoryPerNode, maxSystemMemoryPerNode, maxTotalMemoryPerNode));
> ```

Updated.

**Review by spershin — APPROVED** [review]
Looks good, thanks!

**Pull request merged**